### ISSUE TITLE
Gradle's configuration cache could not be enabled

### ISSUE THREAD

**Issue by danaimset** [LikeTheSalad/android-stem#32]
If project has `org.gradle.unsafe.configuration-cache=true` then build fails:
```
1339 problems were found storing the configuration cache, 50 of which seem unique.
- Task `name` of type `com.likethesalad.tools.resource.locator.android.task.ResourceLocatorTask`: cannot serialize object of type 'com.android.build.gradle.internal.lint.AndroidLintAnalysisTask', a subtype of 'org.gradle.api.Task', as these are not supported with the configuration cache.
... and much more ...
  See https://docs.gradle.org/7.3.3/userguide/configuration_cache.html#config_cache:requirements:task_access
  ```

**Comment by LikeTheSalad** [comment]
Hi @danaimset, I see. I just checked and it seems like [this feature is incubating](https://docs.gradle.org/current/userguide/configuration_cache.html), as of today's latest Gradle version, so it can get major changes on next versions that might break whatever I might do right now to avoid this issue. So I prefer to wait until it's stable.

Also, to be honest the error message you've shared doesn't provide much info on what could be done to avoid it, it mentions that an object of type `'com.android.build.gradle.internal.lint.AndroidLintAnalysisTask'` (which is not a Stem's class but instead comes from the Android Gradle plugin) cannot be serialized. Though there's two things that don't seem to be right, the first one is that there doesn't seem to be such a type defined inside the [ResourceLocatorTask class](https://github.com/LikeTheSalad/resource-locator/blob/main/android-resource-locator/src/main/java/com/likethesalad/tools/resource/locator/android/task/ResourceLocatorTask.kt), and the second one is that the class that cannot be serialized doesn't belong to Stem or its internal tools, but it actually comes from the Android Gradle plugin itself, so I'm left with no clues to get to the root cause of this. Probably the error message is not accurate enough due to the stage of this Gradle's feature, so for now what I an do is wait until it's stable.

**Comment by LikeTheSalad** [comment]
It's been a while now and there's an 8+ version of AGP available, as well as better support in Stem for 8.0.0+ AGP api changes, I'm not sure if the AGP has sorted this issue yet, although this issue has been open for quite some time now and there's no action I can take to sort it from within Stem. So I'm closing this.
